snuba/query/snql/parser.py

```diff
@@ -1347,10 +1349,14 @@ def _post_process(
 
 def parse_snql_query(
     body: str, dataset: Dataset, custom_processing: Optional[CustomProcessors] = None,
-) -> Union[CompositeQuery[QueryEntity], LogicalQuery]:
+) -> Tuple[Union[CompositeQuery[QueryEntity], LogicalQuery], str]:
     with sentry_sdk.start_span(op="parser", description="parse_snql_query_initial"):
         query = parse_snql_query_initial(body)
 
+    with sentry_sdk.start_span(op="parser", description="anonymize_snql_query"):
+        query_copy = copy.deepcopy(query)
+        snql_anonymized = format_snql_anonymized(query_copy).get_sql()
```

Comment:
For reasons I could not figure out, the anonymize formatter modifies the query AST, so a deep copy is required.

Reply:
Nevermind, figured it out: #2284. I'll update this when that is merged.

Reply:
This means that, after #2284  you will remove this change ?

Reply:
I removed the deep copy logic, the anonymizing still happens here